core/server/i18n.js

```diff
@@ -94,23 +121,88 @@ I18n = {
+            i18nTheme = true;
+        }
+
+        // Switch to enable/disable translations of back-end messages from core .js files.
+        // - true: enabled; optional files like content/locales/es.json, etc. are used when available.
+        // - false: disabled; only the default English file core/server/translations/en.json is used.
+        // Before enabling back-end message translations, see previous tasks on issue #6526
+        // (error codes or identifiers, error message translation at the point of display...)
+        backendTranslations = false;
+
+        currentLocale = settingsCache.get('default_locale') || 'en';
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading files for current locale and active theme and keeping their content in memory
+        // (during Ghost's initialization, when English is the language in settings, the second call
+        // to i18n.init() doesn't read again the default English file for core if already in memory).
+        if (i18nCore) {
+            // Reading translation file for core .js files.
+            if (currentLocale === 'en' || !backendTranslations) {
+                blos = fs.readFileSync(path.join(__dirname, 'translations', 'en.json'));
+            } else {
+                // Preventing missing file.
+                try {
+                    blos = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'locales', currentLocale + '.json'));
+                } catch (err) {
+                    if (err.code === 'ENOENT') {
+                        logging.warn('File content/locales/' + currentLocale + '.json not found! Falling back to default core/server/translations/en.json.');
+                        blos = fs.readFileSync(path.join(__dirname, 'translations', 'en.json'));
+                    } else {
+                        throw err;
+                    }
+                }
+            }
+            // if translation file is not valid, you will see an error
+            try {
+                blos = JSON.parse(blos);
+            } catch (err) {
+                blos = undefined;
+                throw err;
+            }
+        }
+        if (i18nTheme && activeTheme) {
+            // Reading translation file for theme .hbs templates.
+            // Compatibility with both old themes and i18n-capable themes.
+            // Falling back any wrong locale to default English for this session.
+            try {
+                blosTheme = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'themes', activeTheme, 'assets', 'locales', currentLocale + '.json'));
+            } catch (err) {
+                blosTheme = undefined;
+                if (err.code === 'ENOENT') {
+                    logging.warn('Theme\'s file assets/locales/' + currentLocale + '.json not found! Falling back to default English locale.');
+                    currentLocale = 'en';
+                    settingsCache.set('default_locale', 'en');
```

Comment:
So if i configure `de` for my blog and i expect the behaviour of the blog to be german, but my current theme has no german translation available, then we revert the blog locale to `en` - can you explain?

Reply:
Yes, not sure which one is the best option for this point. As the code comment says: `// Falling back any wrong locale to default English for this session.`

- If we remove that fall back, then it would be possible to have localization such as for example dates with months in German, and at the same time English theme texts.
- If we keep that fall back, then typos when introducing a wrong locale on settings would result in the default English. However, it's a "just in case" and maybe this is not needed, since there is also the fall back for themes to the default English text within the `{{t}}` translation helpers. And server-side messages go to English as well. Kate tested `moment.locale()` fallbacks in May, and I think they were pretty good. But maybe it's more consistent if all localization goes back to default if translations are missing, instead of a mix of English and e.g. dates in another language.

So, you can say me what's best to do about this for the first version.

Reply:
> If we remove that fall back, then it would be possible to have localization such as for example dates with months in German, and at the same time English theme texts.

Yeah that is correct. If you configure your blog behaviour to german, but your theme is not translated and default english, then it should be your task to fix this.

Reply:
OK, I will remove that locale fall back.

Reply:
One other case to test/cover is. If you have your blog configured to german and theme **does use translation keys**, but not for german, we should fallback to priority english for the theme translation (!) and if english does not exist, it should fallback to the first available locale. The server should print a warning in this case.

Reply:
Well, I think it's a good idea to read the theme's English file if a language is missing, just in case it uses translation keys instead of the easy to read English default. I will add this for themes, without changing the locale. However, the other case of a theme using translation keys instead of texts, and at the same time not even having an English file, would be pretty edge. That should be fixed by the theme designer, in my humble opinion. But as said I will add the fall back for translation keys, a good point.

Reply:
The suggested additional fallback for translation keys added with recent commit. Thanks for pointing it out.

About the other mentioned possibility, although I considered to add it anyway -even being an edge case- using `fs.readdirSync()`, etc., I found that not only it would give an unsolicited -and therefore not useful- non-English language, it also has side effects for the normal case, losing the English default in helpers when not using keys, unless we complicate the code significantly.

So in my opinion the best solution is what is now done, that is the recent commit for a new fallback for translation keys, and the also already added note in the help file suggesting a default English file especially if using keys in helpers instead of the explicit English text, which is logical.